docset/winserver2022-ps/netsecurity/Get-NetFirewallDynamicKeywordAddress.md

```diff
@@ -0,0 +1,198 @@
+---
+description: The Get-NetFirewallDynamicKeywordAddress cmdlet gets dynamic keyword addresses.
+external help file: NetFirewallDynamicKeywordAddress.cmdletDefinition.cdxml-help.xml
+Module Name: NetSecurity
+ms.date: 9/20/2021
+online version: https://docs.microsoft.com/powershell/module/netsecurity/get-netfirewalldynamickeywordaddress?view=windowsserver2022-ps&wt.mc_id=ps-gethelp
+schema: 2.0.0
+title: Get-NetFirewallDynamicKeywordAddress
+---
+
+# Get-NetFirewallDynamicKeywordAddress
+
+## SYNOPSIS
+Gets dynamic keyword addresses.
+
+## SYNTAX
+
+### GetAll (Default)
+```
+Get-NetFirewallDynamicKeywordAddress [-All] [-PolicyStore <String>] [-AllAutoResolve] [-AllNonAutoResolve]
+ [-CimSession <CimSession[]>] [-ThrottleLimit <Int32>] [-AsJob] [<CommonParameters>]
+```
+
+### ById
+```
+Get-NetFirewallDynamicKeywordAddress [-Id] <String[]> [-PolicyStore <String>] [-CimSession <CimSession[]>]
+ [-ThrottleLimit <Int32>] [-AsJob] [<CommonParameters>]
+```
+
+## DESCRIPTION
+The **Get-NetFirewallDynamicKeywordAddress** cmdlet gets dynamic keyword addresses.
+You can specify an ID or get all AutoResolve or all non-AutoResolve dynamic keyword addresses.
+
+## EXAMPLES
+
+### Example 1: Get a dynamic keyword address
+```powershell
+Get-NetFirewallDynamicKeywordAddress Id 258741ed-94a3-406b-9b0c-81fb145a4592 -PolicyStore ContosoPolicyStore
```

Comment:
Is it normal to include a GUID in a parameter for an example command? Copy/pasting this will result in a failure because the customer's computer will not have that rule, but it is not immediately obvious that this should be a template (and they need to use a GUID from an existing rule of their own).

Reply:
I'll replace with "01234567-89ab-cdef-0123-456789abcdef", which is obviously fake. I don't remember why I used a GUID here - if I generated it or took it from an example. (The other approach is to use a place holder "<GUID>, but credible looking input is preferred.)